packages/types/src/Metadata/v2/index.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright 2017-2019 @polkadot/types authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { MetadataInterface } from '../types';
+
+import Option from '../../codec/Option';
+import Struct from '../../codec/Struct';
+import Vector from '../../codec/Vector';
+import Text from '../../primitive/Text';
+import { flattenUniq } from '../util';
+
+import { MetadataCall } from './Calls';
+import { MetadataEvent } from './Events';
+import { MetadataStorage } from './Storage';
+import MetadataRegistry from './MetadataRegistry';
+
+/**
+ * @name MetadataModule
+ * @description
+ * The definition of a module in the system
+ */
+export class MetadataModule extends Struct {
+  constructor (value?: any) {
+    super({
+      name: Text,
+      prefix: Text,
+      storage: Option.with(Vector.with(MetadataStorage)),
+      calls: Option.with(Vector.with(MetadataCall)),
+      events: Option.with(Vector.with(MetadataEvent))
+    }, value);
+  }
+
+  /**
+   * @description the module calls
+   */
+  get calls (): Option<Vector<MetadataCall>> {
+    return this.get('calls') as Option<Vector<MetadataCall>>;
+  }
+
+  /**
+   * @description the module events
+   */
+  get events (): Option<Vector<MetadataEvent>> {
+    return this.get('events') as Option<Vector<MetadataEvent>>;
+  }
+
+  /**
+   * @description the module name
+   */
+  get name (): Text {
+    return this.get('name') as Text;
+  }
+
+  /**
+   * @description the module prefix
+   */
+  get prefix (): Text {
+    return this.get('prefix') as Text;
+  }
+
+  /**
+   * @description the associated module storage
+   */
+  get storage (): Option<Vector<MetadataStorage>> {
+    return this.get('storage') as Option<Vector<MetadataStorage>>;
+  }
+}
+
+/**
+ * @name MetadataV2
+ * @description
+ * The runtime metadata as a decoded structure
+ */
+export default class MetadataV2 extends Struct implements MetadataInterface {
+  constructor (value?: any) {
+    super({
+      modules: Vector.with(MetadataModule),
+      module_path: Text,
```

Comment:
`module_path` is generated from `module_path` macro in the substrate rust code in the runtime module. On substrate node it is `node_runtime`, on substrate template node it can be something else, on parachain it will be whatever module name the chain developer decides.